internal/infrastructure/elastic_search/elastic_search_service.go

```diff
@@ -0,0 +1,113 @@
+package elasticsearch
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/DIMO-Network/device-definitions-api/internal/config"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+)
+
+type ElasticSearch struct {
+	BaseURL        string
+	Token          string
+	MetaEngineName string
+	httpClient     *http.Client
+	log            zerolog.Logger
+}
+
+func NewElasticSearch(settings *config.Settings, logger zerolog.Logger) (*ElasticSearch, error) {
+	var netClient = &http.Client{
+		Timeout: time.Second * 10,
+	}
+	return &ElasticSearch{
+		BaseURL:        settings.ElasticSearchAppSearchHost,
+		Token:          settings.ElasticSearchAppSearchToken,
+		MetaEngineName: "dd-" + settings.Environment,
+		httpClient:     netClient,
+		log:            logger,
+	}, nil
+}
+
+// buildAndExecRequest makes request with token and headers, marshalling passed in obj or if string just passing along in body,
+// and unmarshalling response body to objOut (must be passed in as ptr eg &varName)
+func (d *ElasticSearch) buildAndExecRequest(method, url string, obj interface{}, objOut interface{}) (*http.Response, error) {
```

Comment:
I wonder if we could add something like this to the shared library. There's an HTTP client in there with retry and backoff but it doesn't do marshaling.